File: backend/internal/models/teams.go
package models

import "time"

// Database Models

type Team struct {
	ID             string    `json:"id"`
	OrganisationID string    `json:"organisation_id"`
	Name           string    `json:"name"`
	Description    string    `json:"description"`
	Color          string    `json:"color"`
	CreatedAt      time.Time `json:"created_at"`
}

type TeamMember struct {
	TeamID    string    `json:"team_id"`
	UserID    string    `json:"user_id"`
	CreatedAt time.Time `json:"created_at"`
}

// Data Transfer Objects (for sending data to frontend)

type TeamDTO struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Color       string    `json:"color"`
	MemberCount int16     `json:"member_count"`
	CreatedAt   time.Time `json:"created_at"`
}

type TeamMemberDTO struct {
	User     UserProfile `json:"user"`
	JoinedAt time.Time   `json:"joined_at"` // created_at within db
}

File: frontend/src/api/teams.ts
import { api } from ".";
import type { CustomAppColor } from "../constants/colors";
import type { UserProfile } from "./profiles";

// Models

export type Team = {
    id: string;
    organisation_id: string;
    name: string;
    description: string;
    color: CustomAppColor;
    created_at: string; // ISO string
};

export type TeamMember = {
    user: UserProfile;
    joined_at: string; // ISO string
};

// API Calls

// Fetches all teams of the given organisation
export const fetchTeams = async (org_id: string) => {
    const res = await api.get<Team[]>("/v1/organisations/" + org_id + "/teams");
    return res.data;
};

// Fetches all members of an team
export const fetchTeamMembers = async (org_id: string, team_id: string) => {
    const res = await api.get<TeamMember[]>(
        "/v1/organisations/" + org_id + "/teams" + team_id + "/members"
    );
    return res.data;
};
